### PR TITLE
Add parameterisation

### PR DESCRIPTION
**Pull request by robinmessage**
These commits are fairly separate and individually complete so can be read in order, or as one PR.

### CODE REVIEW

File: src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/containers/ContainerManager.java [hunk 1]
```diff
@@ -231,8 +230,27 @@ public int runSteps(
     return Job.STATUS_OK;
   }
 
+  public ContainerExecResponse runParameterisation(
+      TaskCopy c,
+      File codeDir,
+      RepoInfo repoInfo)
+      throws ApiUnavailableException {
+    if (!c.getDetail().getParameterisation().isPresent()) {
+      return null;
```
Comment (acr31) [inline src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/containers/ContainerManager.java]: Is null the right thing here? Would it be better to return a ContainerExecResponse that shows trivial success?

File: src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/containers/Binding.java [hunk 2]
```diff
@@ -126,6 +123,45 @@ String getMountPoint(String name) {
...
+    }
+
+    @Override
+    ExecutionConfig.Builder applyBinding(ExecutionConfig.Builder builder, String name)
+        throws ContainerExecutionException {
+      if (needsApplying) {
+        needsApplying = false;
+
+        File stepFile = new File(containerTempDir, name);
+        try (FileWriter w = new FileWriter(stepFile)) {
```
Comment (acr31) [inline src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/containers/Binding.java]: Files.asCharsink().write()

Comment (robinmessage) [reply src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/containers/Binding.java]: Done.

File: src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/containers/ContainerManager.java (hunk 1 above)
Comment (robinmessage) [reply src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/containers/ContainerManager.java]: If we return a success it'll get stored as the problemStatement so I'm not sure if it's a good idea. And this is only meant to be called if there is a parameterisation.

I've added a checkNotNull instead as this should only be called if there is a parameterisation. And also on the execution.

Comment (acr31) [reply src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/containers/ContainerManager.java]: ok thanks

**Comment by acr31** [comment]
I'm happy with these changes but lets get the change in to always store problemStatements in the repo too.

**Comment by robinmessage** [comment]
The default problemStatement change is in now.

**Pull request merged**